main/src/ca/uwaterloo/flix/language/phase/Weeder2.scala

```diff
@@ -1261,17 +1261,26 @@ object Weeder2 {
       }
     }
 
+    /**
+      * The condition and then sub-trees are required, but the else sub-tree is optional.
```

Comment:
I think this comment should be moved to the `ifThenElseExpr` parser function. That function should say that there is always a cond/then child but only maybe a else child